support-frontend/app/utils/CheckoutValidationRules.scala

```diff
@@ -34,13 +34,13 @@ object PaidProductValidation {
     SimpleCheckoutFormValidation.passes(createSupportWorkersRequest) &&
     hasValidPaymentDetailsForPaidProduct(createSupportWorkersRequest.paymentFields)
 
-  def hasValidPaymentDetailsForPaidProduct(paymentDetails: PaymentDetails[PaymentFields]): Boolean =
-    paymentDetails match {
-      case PaidProduct(paymentFields) => noEmptyPaymentFields(paymentFields)
-      case FreeProduct => false
-    }
+  def hasValidPaymentDetailsForPaidProduct(paymentDetails: Either[PaymentFields, RedemptionData]): Boolean =
+    paymentDetails fold (
+      paymentFields => noEmptyPaymentFields(paymentFields),
+      _ => false
```

Comment:
now this line is less clear, from the source in Github.  would a pattern match help? or even just name the `_` assuming scalacheck allows that.

Reply:
or even a `// comment`, although again not checked by the compiler

Reply:
or add the `: RedemptionData` which would be type checked I think
```suggestion
      _: RedemptionData => false
```

Reply:
Yes I did think about that but was discouraged by Intellij because it's an 'unused' parameter. I think you're right though and the 'use' is as documentation.